app/models/job.rb

```diff
@@ -10,7 +10,7 @@ class Job < ApplicationRecord
   serialize :context
   alias_attribute :jobid, :guid
 
-  before_destroy :check_active_on_destroy
+  before_destroy :check_active_on_destroy, :prepend => true
```

Comment:
:+1: Looks great, but can a test be added for this change?

Reply:
@bdunne it looks like `before_destroy` was fixed and `throw :abort` does trigger transaction rollback now. Also, there is no reference to objects deleted  by `:dependent => :delete` from inside of `before_destroy`. Not sure how to write test since there is no actual issue as of now.

@bdunne Do you think we do not need this modification (as well as in other models where `before_destroy` present) ?